src/main.rs

```diff
@@ -163,35 +138,69 @@ impl BFInterp {
             MetadataOperand => todo!(),
         };
 
-        let mut string_temp;
-        for arg in call.arguments[1..].iter() {
-            match &arg.0 {
-                LocalOperand { name, .. } => {
-                    string_temp =
-                        string.replacen("{}", self.mem.get(&name).unwrap().to_string().as_str(), 1);
-                    string = &string_temp;
-                }
-                ConstantOperand(con_op) => match con_op.as_ref() {
-                    Constant::Int { value, .. } => {
-                        string_temp = string.replacen("{}", value.to_string().as_str(), 1);
-                        string = &string_temp;
+        let mut string = String::new();
+        let mut arg_it = call.arguments[1..].into_iter();
+        match constant {
```

Comment:
I'm a bit confused. I expected this to be the constant string passed to `printf`, e.g. `1 arg: %i`. Does llvm-ir really turn this into a `Constant::Array` where `elements` is an array for chars?

Reply:
I could only find the value of the global variables through the initializer field which stores the strings as an array of numbers which represent the characters.